src/mixins/java/org/spongepowered/common/mixin/api/mcp/server/level/ServerPlayerMixin_API.java

```diff
@@ -384,6 +390,34 @@ public void setWorldBorder(final @Nullable WorldBorder border) {
 
     // Audience
 
+    @Override
+    @SuppressWarnings("unchecked") // safe casts
+    public <T> Optional<T> get(final Pointer<T> pointer) {
+        if (pointer == Identity.NAME) {
```

Comment:
I'd rather these are just shoved into a map of Pointer to function, having this large amount of if statements is silly.

Reply:
tbh I'd like to see this become a switch expression once Java supports those properly, but for now I'm not sure I see the benefit of a map *unless* the set of supported pointers is extendable.

Reply:
I see no reason for a map of functions to exist - that is just going to introduce unnecessary complexity here. There is **far** worse code in the repository.

Reply:
Well sure, there might be more pointers in the future, but like from the standpoint of "if your if else is more than three, then it's silly to make it an if else...

Reply:
I don't expect that there will be more pointers though because these are generally used in _identity_ and there really doesn't need to be any more.

For what it's worth, if this does expand - from a performance point of view, a hash map may be better. For a four item if statement - I don't think it matters. Personally, if I was to do this I'd have done a map because that's how we did it in some of our previous professional projects and it made it super easy to extend. It's not that complicated but it might not be obvious to young developers. Java doing switch properly would put all of this to rest because then it'll decide the right way to do it.

An if statement is fine here. It's not like this is going to get hit all that often, if at all really, and I don't see it being extended, as I said.

> There is far worse code in the repository.

As a note, we shouldn't be trying to race to the bottom as a baseline for the standards we should uphold. This code isn't actually bad, but we should _never_ justify a design decision based on the worst in the repo. It's a big project, there is bad code. Readable, or at least commented (especially if it's "clever") code is what we aim for here.